File: 13.cpp
#include <iostream>
#include <string>

using namespace std;

template <typename TFirst, typename TSecond>
class Pair {
public:
		TFirst First;
		TSecond Second;
		Pair(int x, int y) {
				this->First = x;
				this->Second = y;
		}
};

int main() {
		Pair<int, int>* p1 = new Pair<int, int>(0, 0);
		printf("Pair`2[System.Int32,System.Int32]\n");
		printf("%d %d\n", p1->First, p1->Second);
}

File: 13.cs
using System;

public class Pair<TFirst,TSecond> {
				public TFirst First;
				public TSecond Second;
				public Pair(TFirst x, TSecond y) {
								First = x;
								Second = y;
				}

}

public class MainRunner {
				static void Main() {
								Pair<int, int> p1 = new Pair<int, int>(0, 0);
								Console.WriteLine(p1);
								Console.WriteLine(p1.First + " " + p1.Second);
				}
}